### PR TITLE
[PTQ] Metatypes with bias list alignment

### PR DESCRIPTION
**Pull request by KodiaqQ**
### Changes

- Reduced metatypes in `OPERATIONS_WITH_BIAS` list for ONNX;
- Reduced metatypes in `OPERATIONS_WITH_BIAS` list for PyTorch;
- Separated GroupConvolution from the Convolutions on ONNX;

### Reason for changes

- Alignment for algorithms between backends;

### Related tickets

- 133198
- 104166
- 106469

### Tests

- post_training_quantization/323 - failed with regressions for ONNX (as expected)
- post_training_quantization/324 - passed (updated references)

### CODE REVIEW

File: nncf/torch/graph/operator_metatypes.py [hunk 1]
```diff
@@ -1038,12 +1038,7 @@ def get_operator_metatypes() -> List[Type[OperatorMetatype]]:
     PTModuleConv1dMetatype,
     PTModuleConv2dMetatype,
     PTModuleConv3dMetatype,
-    PTDepthwiseConv1dSubtype,
-    PTDepthwiseConv2dSubtype,
-    PTDepthwiseConv3dSubtype,
-    PTModuleConvTranspose1dMetatype,
```
Comment (alexsu52) [inline nncf/torch/graph/operator_metatypes.py:1044]: It looks like `ConvTranspose` operations have similar semantic as `Convolution`. From BC perspective, what is strategy for it? Do you have a list of models where `ConvTranspose` operations are used?

Useful link: https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html

Comment (KodiaqQ) [reply nncf/torch/graph/operator_metatypes.py:1044]: Since we have only conformance models scope for the PyTorch backend, the removal of the **ConvTranspose** doesn't affect the accuracy for any model here. So, we can decide that the models from the conformance scope don't contain **ConvTranspose** or this layer didn't work anyway, and was added with no reason and testing.
@AlexanderDokuchaev, could you please clarify?

Comment (AlexanderDokuchaev) [reply nncf/torch/graph/operator_metatypes.py:1044]: What i should clarify? I have no list of used modules in test.
`ConvTranspose` was added as operation with bias...
But as i remember transpose convolution in OV has type `ConvolutionBackpropData`, that does not exist in OPERATIONS_WITH_BIAS for OV backend.

So, if transposed layers should not be supported it `ConvTranspose` can be removed.  In otherwise need to check that some model using `ConvTranspose` and add this kind of model to test scope if need.

And looks like `OPERATIONS_WITH_BIAS` is not correct name to current use, may be it should be `metatypes_for_fast_bias_correction`.

File: nncf/onnx/graph/metatypes/onnx_metatypes.py [hunk 2]
```diff
@@ -711,12 +745,9 @@ def _is_depthwise_conv(model: onnx.ModelProto, node: onnx.NodeProto) -> bool:
     :param node: Convolution node to check whether it is depthwise.
     :return: True if the convolution is depthwise, False - otherwise.
     """
-    conv_group = None
```
Comment (AlexanderDokuchaev) [inline nncf/onnx/graph/metatypes/onnx_metatypes.py:714]: After changes `conv_group` used while definition is under if condition, as result possible that `conv_group` uses without definition and raise error.

File: nncf/openvino/graph/metatypes/groups.py [hunk 3]
```diff
@@ -185,7 +185,6 @@
 # Contains the operation metatypes for which bias can be applied.
 OPERATIONS_WITH_BIAS = [
     ov_metatypes.OVConvolutionMetatype,
-    # TODO: add all metatypes with bias
```
Comment (AlexanderDokuchaev) [inline nncf/openvino/graph/metatypes/groups.py:188]: Was added all metatypes with bias?

File: nncf/onnx/graph/metatypes/onnx_metatypes.py [hunk 4]
```diff
@@ -85,6 +86,22 @@ def matches(cls, model: onnx.ModelProto, node: onnx.NodeProto) -> bool:
         return _is_depthwise_conv(model, node)
 
 
+@ONNX_OPERATION_METATYPES.register()
+class ONNXGroupConvolutionMetatype(ONNXOpWithWeightsMetatype):
```
Comment (AlexanderDokuchaev) [inline nncf/onnx/graph/metatypes/onnx_metatypes.py:90]: Should be added groupconv metatypes in PT backend?

File: nncf/torch/graph/operator_metatypes.py (hunk 1 above)
Comment (KodiaqQ) [reply nncf/torch/graph/operator_metatypes.py:1044]: > What i should clarify?

Need to clarify why you've added these layers to the **OPERATORS_WITH_BIAS_METATYPES** list. **ConvolutionBackpropData** wasn't added to the OV backend for a reason: it possibly causes accuracy degradation.

File: nncf/openvino/graph/metatypes/groups.py (hunk 3 above)
Comment (KodiaqQ) [reply nncf/openvino/graph/metatypes/groups.py:188]: No. And they will not.

File: nncf/onnx/graph/metatypes/onnx_metatypes.py (hunk 4 above)
Comment (KodiaqQ) [reply nncf/onnx/graph/metatypes/onnx_metatypes.py:90]: If it is possible, we may extend metatypes in the PT backend. But not in this PR, I hope.

File: nncf/onnx/graph/metatypes/onnx_metatypes.py (hunk 2 above)
Comment (KodiaqQ) [reply nncf/onnx/graph/metatypes/onnx_metatypes.py:714]: Since **ONNXDepthwiseConvolutionMetatype** became the subtype of the **ONNXGroupConvolutionMetatype** that must have a _group_ attribute, it's not possible that _conv_group_ wouldn't be found.

File: nncf/onnx/graph/metatypes/onnx_metatypes.py (hunk 4 above)
Comment (AlexanderDokuchaev) [reply nncf/onnx/graph/metatypes/onnx_metatypes.py:90]: 135885

File: nncf/torch/graph/operator_metatypes.py (hunk 1 above)
Comment (alexsu52) [reply nncf/torch/graph/operator_metatypes.py:1044]: I hope, the comment will be resolved in 106469.

**Review by alexsu52 — APPROVED** [review]
LGTM

**Comment by KodiaqQ** [comment]
@AlexanderDokuchaev, @kshpv, please, review.

**Review by kshpv — APPROVED** [review]

**Review by AlexanderDokuchaev — APPROVED** [review]
